lib/routemaster/middleware/payload_filter.rb

```diff
@@ -0,0 +1,10 @@
+module Routemaster
+  module Middleware
+    class PayloadFilter
+      # Filters duplicate events by url and type in a single payload.
+      def run(payload)
+        payload.group_by { |event| [event['url'], event['type']] }.map { |_, events| events.last }
```

Comment:
Probably does make sense to return the latest `t` though, just to avoid any confusion.